File: FirServer/FirServer/Common/BaseBehaviour.cs
using FirServer.Interface;
using FirServer.Manager;

namespace FirServer.Common
{
    public class BaseBehaviour : IObject
    {
        private static ConfigManager? _configMgr = null;
        protected static ConfigManager? configMgr
        {
            get
            {
                if (_configMgr == null)
                {
                    _configMgr = ManagementCenter.GetManager<ConfigManager>();
                }
                return _configMgr;
            }
        }

        private static AssemblyManager? _assemblyMgr = null;
        protected static AssemblyManager? assemblyMgr
        {
            get
            {
                if (_assemblyMgr == null)
                {
                    _assemblyMgr = ManagementCenter.GetManager<AssemblyManager>();
                }
                return _assemblyMgr;
            }
        }

        private static NetworkManager? _netMgr = null;
        protected static NetworkManager? netMgr
        {
            get
            {
                if (_netMgr == null)
                {
                    _netMgr = ManagementCenter.GetManager<NetworkManager>();
                }
                return _netMgr;
            }
        }

        private static HandlerManager? _handlerMgr = null;
        protected static HandlerManager? handlerMgr
        {
            get
            {
                if (_handlerMgr == null)
                {
                    _handlerMgr = ManagementCenter.GetManager<HandlerManager>();
                }
                return _handlerMgr;
            }
        }

        private static WorldManager? _worldMgr = null;
        protected static WorldManager? worldMgr
        {
            get
            {
                if (_worldMgr == null)
                {
                    _worldMgr = ManagementCenter.GetManager<WorldManager>();
                }
                return _worldMgr;
            }
        }

        private static LoggerManager? _loggerMgr = null;
        protected static LoggerManager? loggerMgr
        {
            get
            {
                if (_loggerMgr == null)
                {
                    _loggerMgr = ManagementCenter.GetManager<LoggerManager>();
                }
                return _loggerMgr;
            }
        }

        private static DataManager? _dataMgr = null;
        protected static DataManager? dataMgr
        {
            get
            {
                if (_dataMgr == null)
                {
                    _dataMgr = ManagementCenter.GetManager<DataManager>();
                }
                return _dataMgr;
            }
        }

        private static TimerManager? _timerMgr = null;
        protected static TimerManager? timerMgr
        {
            get
            {
                if (_timerMgr == null)
                {
                    _timerMgr = ManagementCenter.GetManager<TimerManager>();
                }
                return _timerMgr;
            }
        }

        private static ModelManager? _modelMgr = null;
        protected static ModelManager? modelMgr
        {
            get
            {
                if (_modelMgr == null)
                {
                    _modelMgr = ManagementCenter.GetManager<ModelManager>();
                }
                return _modelMgr;
            }
        }
    }
}

File: FirServer/FirServer/src/com/common/BaseBehaviour.java
package com.common;

import com.manager.*;

public class BaseBehaviour {
    private static ConfigManager _configMgr = null;
    protected static ConfigManager configMgr()
    {
        if (_configMgr == null)
        {
            _configMgr = (ConfigManager)MgrCenter.GetManager(ConfigManager.class);
        }
        return _configMgr;
    }

    private static NetworkManager _netMgr = null;
    protected static NetworkManager netMgr()
    {
        if (_netMgr == null)
        {
            _netMgr = (NetworkManager)MgrCenter.GetManager(NetworkManager.class);
        }
        return _netMgr;
    }

    private static HandlerManager _handlerMgr = null;
    protected static HandlerManager handlerMgr()
    {
        if (_handlerMgr == null)
        {
            _handlerMgr = (HandlerManager)MgrCenter.GetManager(HandlerManager.class);
        }
        return _handlerMgr;
    }

    private static TableManager _tableMgr = null;
    protected static TableManager tableMgr()
    {
        if (_tableMgr == null)
        {
            _tableMgr = (TableManager)MgrCenter.GetManager(TableManager.class);
        }
        return _tableMgr;
    }

    private static ModelManager _modelMgr = null;
    protected static ModelManager modelMgr()
    {
        if (_modelMgr == null)
        {
            _modelMgr = (ModelManager)MgrCenter.GetManager(ModelManager.class);
        }
        return _modelMgr;
    }

    private static LogManager _logMgr = null;
    protected static LogManager logMgr()
    {
        if (_logMgr == null)
        {
            _logMgr = (LogManager)MgrCenter.GetManager(LogManager.class);
        }
        return _logMgr;
    }

    private static DataManager _dataMgr = null;
    protected static DataManager dataMgr()
    {
        if (_dataMgr == null)
        {
            _dataMgr = (DataManager)MgrCenter.GetManager(DataManager.class);
        }
        return _dataMgr;
    }
}
